CRISPResso2/CRISPRessoShared.py

```diff
@@ -1968,53 +1969,524 @@ def check_custom_config(args):
+
+        Parameters:
+        -----------
+        total_reads : int
+            Total reads, unaligned and aligned
+        n_read_aligned : int
+            Total aligned reads
+        """
+        if total_reads == 0:
+            return
+        if (n_read_aligned/total_reads) <= self.cutoff:
+            self.message = self.message + " Total reads: {}, Aligned reads: {}".format(total_reads, n_read_aligned)
+            self.messageHandler.display_warning(self.message)
+            self.messageHandler.report_warning(self.message)
+
+
+class LowReadsAlignedToAmpliconGuardRail:
+    """Guardrail class: check if the distribution of reads is roughly even across amplicons"""
+    def __init__(self, messageHandler, cutoff):
+        """Assign variables and create guardrail message
+
+        Parameters:
+        -----------
+        messageHandler : GuardRailMessagehandler
+            Guardrail message handler to create and display warnings
+        cutoff : float
+            The float representation of the accepted percentage deviation down from the expected distribution
+        """
+        self.messageHandler = messageHandler
+        self.message = " <={val}% of expected reads were aligned to amplicon: ".format(val=(cutoff * 100))
+        self.cutoff = cutoff
+
+    def safety(self, total_reads, reads_aln_amplicon):
+        """Safety check, if total_reads divided by n_reads_aligned is lower than the total_reads divided by the number of amplicons
+
+        Parameters:
+        -----------
+        total_reads : int
+            Total reads, unaligned and aligned
+        reads_aln_amplicon : dict
+            A dictionary with the names of amplicons as the key and the number of reads aligned as the value
+        """
+        expected_per_amplicon = total_reads / len(reads_aln_amplicon.keys())
+        for amplicon, aligned in reads_aln_amplicon.items():
+            if aligned <= (expected_per_amplicon * self.cutoff):
+                amplicon_message = self.message + amplicon + ", Percent aligned: {}".format((aligned/total_reads) * 100)
+                self.messageHandler.display_warning(amplicon_message)
+                self.messageHandler.report_warning(amplicon_message)
+
+
+class HighReadsAlignedToAlternateAmpliconGuardRail:
+    """Guardrail class: check if the distribution of reads is roughly even across amplicons"""
+    def __init__(self, messageHandler, cutoff):
+        """Assign variables and create guardrail message
+
+        Parameters:
+        -----------
+        messageHandler : GuardRailMessagehandler
+            Guardrail message handler to create and display warnings
+        cutoff : float
+            The float representation of the accepted percentage deviation up from the expected distribution
+        """
+        self.messageHandler = messageHandler
+        self.message = " >={val}% more reads than expected were aligned to amplicon: ".format(val=(cutoff * 100))
```

Comment:
Should we replace `>=` with `≥`?

Reply:
Here, and other places

Reply:
Before we make this change we should make sure that this character can be printed in the terminal. It works on my terminals, but we should test it on others as well.

Reply:
The more I'm looking into this, it seems like a rabbit hole... let's leave it as `>=`.